src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -838,28 +840,35 @@ private List<GrantedAuthority> determineAuthorities(IdToken idToken, GenericJson
         grantedAuthorities.add(SecurityRealm.AUTHENTICATED_AUTHORITY2);
 
         if (isNotBlank(groupsFieldName)) {
-            if (!Strings.isNullOrEmpty(userInfoServerUrl) && containsField(userInfo, groupsFieldName)) {
-                LOGGER.fine("UserInfo contains group field name: " + groupsFieldName + " with value class:" + getField(userInfo, groupsFieldName).getClass());
-                List<String> groupNames = ensureString(getField(userInfo, groupsFieldName));
+            // if groupsFieldName contains []., then groupsFieldName
+            // is first portion, and nestedGroupFieldName is
+            // second portion
+            String[] parts = groupsFieldName.split("\\[\\]\\.");
+            // don't modify groupsFieldName
+            String localGroupsFieldName = parts.length > 1 ? parts[0] : groupsFieldName;
```

Comment:
If there is no split, parts[0] will be the content of groupsFieldName, conditional is not required

Reply:
updated in latest changes